src/views/home/components/transaction-form/transaction-form.view.tsx

```diff
@@ -7,66 +9,44 @@ import Typography from "src/views/shared/typography/typography.view";
 import Card from "src/views/shared/card/card.view";
 import TokenIcon from "src/views/shared/token-icon/token-icon.view";
 import List from "src/views/home/components/list/list.view";
-import { useUIContext } from "src/contexts/ui.context";
 import tokens from "src/assets/tokens/tokens.json";
+import Button from "src/views/shared/button/button.view";
+import AmountInput from "src/views/home/components/amount-input/amount-input.view";
 import { chains } from "src/constants";
 import { useTransactionContext } from "src/contexts/transaction.context";
 import { Chain, Token, TransactionData } from "src/domain";
-import Button from "src/views/shared/button/button.view";
 
 const defaultTransaction: TransactionData = {
   from: chains[0],
   to: chains[1],
   token: tokens[0],
-  amount: 0,
+  amount: BigNumber.from(0),
 };
 
 const TransactionForm: FC = () => {
   const classes = useTransactionFormtStyles();
-  const { openModal, closeModal } = useUIContext();
+  const [list, setList] = useState<List>();
   const { setTransaction } = useTransactionContext();
+  const [isInvalid, setIsInvalid] = useState(true);
   const [localTransaction, setLocalTransaction] = useState(defaultTransaction);
   const ChainFromIcon = localTransaction.from.icon;
   const ChainToIcon = localTransaction.to.icon;
 
   const onChainFromButtonClick = (from: Chain) => {
     setLocalTransaction({ ...localTransaction, from });
-    closeModal();
+    setList(undefined);
   };
   const onChainToButtonClick = (to: Chain) => {
     setLocalTransaction({ ...localTransaction, to });
-    closeModal();
+    setList(undefined);
   };
   const onTokenClick = (token: Token) => {
     setLocalTransaction({ ...localTransaction, token });
-    closeModal();
-  };
-  const componentFromChain = () => (
-    <List
-      placeholder="Search Network"
-      list={{ type: "chain", items: chains, onClick: onChainFromButtonClick }}
-    />
-  );
-  const componentToChain = () => (
-    <List
-      placeholder="Search Network"
-      list={{ type: "chain", items: chains, onClick: onChainToButtonClick }}
-    />
-  );
-  const componentToken = () => (
-    <List
-      placeholder="Search token"
-      list={{ type: "token", items: tokens, onClick: onTokenClick }}
-    />
-  );
-  const onChainFromSelectorClick = () => {
-    openModal(componentFromChain);
-  };
-  const onChainToSelectorClick = () => {
-    openModal(componentToChain);
+    setList(undefined);
   };
-  const onTokenSelectorClick = () => {
-    openModal(componentToken);
+  const onChange = ({ amount, isInvalid }: { amount: BigNumber; isInvalid: boolean }) => {
```

Comment:
We agreed not to use inline types.

I would allow it, but it was decided otherwise. Do you want to reopen the discussion?